.github/workflows/python-publish.yaml

```diff
@@ -15,18 +15,20 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
       with:
-        python-version: '3.8'
+        python-version: '3.12'
+
+    - name: Update PATH
+      run: 'echo "$HOME/.local/bin" >> "$GITHUB_PATH"'
 
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
-        python -m pip install setuptools==47.3.0 wheel==0.34.2 twine==3.1.1 requests==2.29.0
+        pipx install poetry==1.8.2
+        pipx install twine
```

Comment:
I am not 100% sure this is correct but I believe `pipx` is installed by `setup-python`. This is difficult to test, so I'll pay special attention to this during the next release.

Reply:
probably does but python -m pip works no matter what! why not stick to it?

equivalent in js:

using npm to install npx to install yarn.

Reply:
It's not guaranteed to work if there are conflicting packages already installed in global/user site-packages. pipx gives Poetry an isolated virtualenv which is why I like it for stuff like this, though probably not a huge deal for this particular workflow, especially if we get rid of twine.